pkg/resources/grant_helpers.go

```diff
@@ -54,12 +55,20 @@ type grant struct {
 // splitGrantID takes the <db_name>|<schema_name>|<view_name>|<privilege> ID and
 // returns the object name and privilege.
 func splitGrantID(v string) (string, string, string, string, error) {
-	arr := strings.Split(v, "|")
-	if len(arr) != 4 {
+
+	reader := csv.NewReader(strings.NewReader(v))
+	reader.Comma = '|'
+
+	lines, err := reader.ReadAll()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
log.Fatal will call os.Exit, which we don't want. splitGrant allows you to return an error, so you could do `return "", "", "", "", err` and let the caller deal with the issue instead of crashing the program